libs/functional/CMakeLists.txt

```diff
@@ -21,14 +21,17 @@ set(functional_headers
   hpx/functional/bind.hpp
   hpx/functional/bind_back.hpp
   hpx/functional/bind_front.hpp
+  hpx/functional/first_argument.hpp
   hpx/functional/function.hpp
   hpx/functional/function_ref.hpp
   hpx/functional/invoke.hpp
   hpx/functional/invoke_fused.hpp
   hpx/functional/mem_fn.hpp
   hpx/functional/one_shot.hpp
   hpx/functional/protect.hpp
+  hpx/functional/reinitializable_static.hpp
   hpx/functional/result_of.hpp
+  hpx/functional/static_reinit.hpp
```

Comment:
`first_argument` fits here nicely but `reinitializable_static` and `static_reinit` not so much. Maybe add them to a module `hpx_static_reinit`? `hpx_reinitialiazable_static` is a bit of a mouthful but also works. `hpx_static` could have too many other meanings...